packages/reference/src/components/LinkActions/helpers.ts

```diff
@@ -0,0 +1,75 @@
+import { Asset, EntityType, Entry, FieldExtensionSDK } from '../../types';
+import { ReferenceValidations } from '../../utils/fromFieldValidations';
+
+export async function createEntity(props: {
+  sdk: FieldExtensionSDK;
+  entityType: EntityType;
+  contentTypeId?: string;
+}) {
+  if (props.entityType === 'Entry') {
+    if (!props.contentTypeId) {
+      return {};
+    }
+    const { entity, slide } = await props.sdk.navigator.openNewEntry<Entry>(props.contentTypeId, {
+      slideIn: true,
+    });
+    return { entity, slide };
+  } else {
+    const { entity, slide } = await props.sdk.navigator.openNewAsset<Asset>({
+      slideIn: true,
+    });
+    return { entity, slide };
+  }
+}
+
+export async function selectSingleEntity(props: {
+  sdk: FieldExtensionSDK;
+  entityType: EntityType;
+  validations: ReferenceValidations;
+}) {
+  if (props.entityType === 'Entry') {
+    return await props.sdk.dialogs.selectSingleEntry<Entry>({
+      locale: props.sdk.field.locale,
+      contentTypes: props.validations.contentTypes,
+    });
+  } else {
+    return props.sdk.dialogs.selectSingleAsset<Asset>({
+      locale: props.sdk.field.locale,
+      mimetypeGroups: props.validations.mimetypeGroups,
+    });
+  }
+}
+
+export async function selectMultipleEntities(props: {
+  sdk: FieldExtensionSDK;
+  entityType: EntityType;
+  validations: ReferenceValidations;
+}) {
+  const value = props.sdk.field.getValue();
+
+  const linkCount = Array.isArray(value) ? value.length : value ? 1 : 0;
+
+  // TODO: Why not always set `min: 1` by default? Does it make sense to enforce
+  //  user to select as many entities as the field's "min" requires? What if e.g.
+  // "min" is 4 and the user wants to insert 2 entities first, then create 2 new ones?
+  const min = Math.max((props.validations.numberOfLinks?.min || 1) - linkCount, 1);
```

Comment:
@DanweDE @djagya maybe it's time to clean this `min/max` complications? I think a lot of customers were complaining about it

Reply:
I thought I recently updated the TODO but apparently, I didn't :thinking: 

I don't think we should be passing `1`. Instead, `sdk.dialogs.selectMultipleEntries` should always allow inserting an arbitrary number of links (while potentially informing you about limits by looking at these `min`/`max` values). What do you think?

Reply:
Yeah, it looks logical to me. I can incorporate this logic into the new entity editor, instead of disabling the button I would just inform the use about the restrictions.

Reply:
That would be grand :+1:

Reply:
Deal 👍

Reply:
So what's the plan? Do I remove anything here now?

Reply:
No, you can leave it as it is. Maybe the comment is non needed anymore though